cluster-autoscaler/core/scaledown/actuation/actuator.go

```diff
@@ -197,10 +204,12 @@ func (a *Actuator) taintNodesSync(NodeGroupViews []*budgets.NodeGroupView) error
 		updateLatencyTracker.AwaitOrStopChan <- true
 		latency, ok := <-updateLatencyTracker.ResultChan
 		if ok {
+			a.pastLatencies.RegisterElement(latency)
+			a.pastLatencies.DropNotNewerThan(time.Now().Add(-1 * pastLatencyExpireDuration))
 			// CA is expected to wait 3 times the round-trip time between CA and the api-server.
-			// Therefore, the nodeDeleteDelayAfterTaint is set 2 times the latency.
-			// A delay of one round trip time is implicitly there when measuring the latency.
-			a.nodeDeleteDelayAfterTaint = 2 * latency
+			// At this point, we have already tainted all the nodes.
+			// Therefore, the nodeDeleteDelayAfterTaint is set 2 times the maximum latency observed during the last hour.
```

Comment:
Not directly related to this change, but is this actually correct? `nodeDeleteDelayAfterTaint` is written here and read from other goroutines, so perhaps it should be passed explicitly instead of used as a field on the actuator object?

Reply:
Done